### PR TITLE
8336679: Add @implSpec for the default implementations in Process.waitFor()

### PR DESCRIPTION
**Pull request by naotoj**
This is a simple doc-only change that follows up [JDK-8336479](https://bugs.openjdk.org/browse/JDK-8336479). A corresponding CSR has also been drafted.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8336936](https://bugs.openjdk.org/browse/JDK-8336936) to be approved

### Issues
 * [JDK-8336679](https://bugs.openjdk.org/browse/JDK-8336679): Add @implSpec for the default implementations in Process.waitFor() (**Bug** - P4)
 * [JDK-8336936](https://bugs.openjdk.org/browse/JDK-8336936): Add @implSpec for the default implementations in Process.waitFor() (**CSR**)

### Reviewers
 * [Brian Burkhalter](https://openjdk.org/census#bpb) (@bplb - **Reviewer**)
 * [Justin Lu](https://openjdk.org/census#jlu) (@justin-curtis-lu - Committer) 🔄 Re-review required (review applies to [bd91aba2](https://git.openjdk.org/jdk/pull/20285/files/bd91aba24f315749deef6a6b6a5a3ebe44d95ea7))
 * [Chen Liang](https://openjdk.org/census#liach) (@liach - **Reviewer**) 🔄 Re-review required (review applies to [17c61d99](https://git.openjdk.org/jdk/pull/20285/files/17c61d99d50b8759fa691f2fbcc811716d6ca4e8))

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20285/head:pull/20285` \
`$ git checkout pull/20285`

Update a local copy of the PR: \
`$ git checkout pull/20285` \
`$ git pull https://git.openjdk.org/jdk.git pull/20285/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20285`

View PR using the GUI difftool: \
`$ git pr show -t 20285`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20285.diff">https://git.openjdk.org/jdk/pull/20285.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20285#issuecomment-2243902774)

### CODE REVIEW

**Review by bplb — APPROVED** [review]

**Review by justin-curtis-lu — APPROVED** [review]

File: src/java.base/share/classes/java/lang/Process.java [hunk 1]
```diff
@@ -442,10 +442,12 @@ public final BufferedWriter outputWriter(Charset charset) {
...
-     * <p>The default implementation of this method polls the {@code exitValue}
-     * to check if the process has terminated. Concrete implementations of this
-     * class are strongly encouraged to override this method with a more
-     * efficient implementation.
+     * @implSpec
+     * The default implementation of this method polls the {@code exitValue}
+     * to check if the process has terminated.
+     * @implNote
+     * Concrete implementations of this class are strongly encouraged to
+     * override this method with a more efficient implementation.
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/lang/Process.java:450]:
```suggestion
     * @implSpec
     * The default implementation of this method polls the {@code exitValue}
     * to check if the process has terminated.
     *
     * @implNote
     * Concrete implementations of this class are strongly encouraged to
     * override this method with a more efficient implementation.
```

File: src/java.base/share/classes/java/lang/Process.java [hunk 2]
```diff
@@ -486,10 +488,12 @@ public boolean waitFor(long timeout, TimeUnit unit)
...
```
Comment (ExE-Boss) [inline src/java.base/share/classes/java/lang/Process.java:496]:
```suggestion
     * @implSpec
     * The default implementation of this method polls the {@code exitValue}
     * to check if the process has terminated.
     *
     * @implNote
     * Concrete implementations of this class are strongly encouraged to
     * override this method with a more efficient implementation.
```

**Review by ExE-Boss — COMMENTED** [review]
Maybe add an empty line between `@implSpec` and `@implNote`:

**Review by liach — APPROVED** [review]
Thank you for addressing my CSR comment about subclass recommendations.

**Review by bplb — APPROVED** [review]
Re-reviewed.

**Comment by naotoj** [comment]
Thanks for the reviews & suggestions!
/integrate

**Review by bplb — APPROVED** [review]

**Pull request closed without merging**